server/app/controllers/applicant/ApplicantRoutes.java

```diff
@@ -0,0 +1,41 @@
+package controllers.applicant;
+
+import auth.CiviFormProfile;
+import auth.ProfileFactory;
+import play.api.mvc.Call;
+
+/**
+ * Class that computes routes for applicant actions. The route for an applicant may be different
+ * from that for a TI taking action on behalf of an applicant.
+ */
+public final class ApplicantRoutes {
+
+  // There are two cases where we want to use the URL that contains the applicant id:
+  // - TIs performing actions on behalf of applicants.
+  // - The applicant has a profile that does not (yet) include the applicant id.
+  //   This case will eventually go away once existing profiles have expired and been replaced.
+  private static boolean includeApplicantIdInRoute(CiviFormProfile profile) {
+    boolean isTi = profile.isTrustedIntermediary();
+    // TODO(yotommy): add a counter to monitor this case
```

Comment:
suggestion (non-blocking): do this now, now that the counter PR is ready? otherwise file an issue for it `TODO([issue #])`

Reply:
Done, and simpler now that we are using instance methods rather than static methods.